arangod/Cluster/SynchronizeShard.cpp

```diff
@@ -951,4 +953,15 @@ bool SynchronizeShard::first() {
 
 }
 
-SynchronizeShard::~SynchronizeShard() {}
+
+void SynchronizeShard::setState(ActionState state) {
```

Comment:
See notes on same routine in CreateShard.  incShardVersion() becomes unnecessary if hash changed during transition to final state.  Which has to precede notify() call.

Reply:
Again, please let's postpone this for after GA. I make a waffle issue.